src/audit/utils.ts

```diff
@@ -211,6 +240,81 @@ const metricPaths = [
   nodeConnectionOutboundMetricPath,
 ] as const;
 
+/**
+ * Will take an array of dot path sorted paths and return the minimal list common paths.
+ * So sub-paths will be filtered out if we already contain a parent path E.G. `a.b` will be removed if we also include `a`.
+ * Duplicate paths will be removed, so `a` will be removed if two `a`'s exist.
+ */
+function filterSubPaths(paths: Array<string>): Array<string> {
+  let previous: string = '';
+  return paths.sort().filter((value, index) => {
+    // Checking if the current value is included within the previous
+    if (index === 0 || !value.startsWith(previous)) {
+      previous = value;
+      return true;
+    }
+    return false;
+  }, {});
+}
+
+/**
+ * This takes N generators that yield data in a sorted order and combines their outputs in a fully sorted order.
+ * This will only work on pre-sorted outputs from the generator.
+ */
+async function* genSort<T>(
+  sortFn: (a: T, b: T) => number,
+  ...gens: Array<AsyncGenerator<T, void, void>>
+): AsyncGenerator<T, void, void> {
```

Comment:
This is pretty generic, We could make it a utility in the top level domain. Not sure.

Reply:
I'm going to leave it as is, If we need this functionality elsewhere then we can move it.